### DISCUSSION TITLE
Position of an active MRTK 3 pointer

### DISCUSSION THREAD

**Discussion by camroo29** [microsoft/MixedRealityToolkit-Unity#11645] [Q&A]
Hi, 

Since MRTK 3 architecture has been redesigned based on XRI, porting MRTK 2 application is a little confusing especially how to get the coordinates of an active Pointer or controller. Any sample code or example scenes (similar to MRTK2/Examples/Demos/Input) would be really helpful.

Thanks

**Comment by AMollis** [comment]
@camroo29 ... moving this to our discussion forum 

Can you please describe your scenario?  We have many MRTK3 sample scenes already, one of these might demonstrate what you need.

We also have this migration document, which might help a bit:

https://learn.microsoft.com/en-us/windows/mixed-reality/mrtk-unity/mrtk3-overview/architecture/mrtk-v2-to-v3

**Comment by Ram1508** [comment]
Hi @AMollis, I have a similar situation:

I am trying to spawn an object in space where the pointer is being clicked. Is there any Api  to get the world position of where pointer is air tapped, or Examples scenes to be explored in MRTK3.

**Comment by camroo29** [comment]
@AMollis I am trying to get the coordinates of the pointer (when there is any raycast hit) and the fingertip to find the distance between two points. In MRTK 2, ~/Examples/Demos/Input had an example for finding the fingertip and for getting the pointer coordinates:

```csharp
    void Update()
    {
        foreach(var source in MixedRealityToolkit.InputSystem.DetectedInputSources)
        {
            if (source.SourceType == Microsoft.MixedReality.Toolkit.Input.InputSourceType.Hand)
            {
                foreach (var p in source.Pointers)
                {
                    if (p is IMixedRealityNearPointer)
                    {
                         continue;
                    }
                    if (p.Result != null)
                    {
                        var startPoint = p.Position;
                        var endPoint = p.Result.Details.Point;
                        var hitObject = p.Result.Details.Object;
                    }
                }
            }
        }
    }
```
Some examples of using Microsoft.MixedReality.Toolkit.Input and Interactors for MRTK 3 would be great.
